README.md

```diff
@@ -1,59 +1,85 @@
+The core component of Guard is the [guard-gate](pkg/guard-gate) which can be used to extend Knative Queue Proxy. [guard-gate](pkg/guard-gate) monitors and potentially blocks requests and/or responses to Knative services based on a per-service security configuration.
 
-Overall the solution offers both visibility into the security of the service as well as the ability to monitor/block both known patterns and unknown patterns (using a zero day exploits).
+The per service security configuration is stored in a **"Guardian"** object. **Guardian** maintains a set of micro-rules that enable a fine grain filtering, performed against each value delivered to/from the service.
+By using the micro-rules, [guard-gate](pkg/guard-gate) can identify the delivery of exploits targeting vulnerabilities embedded as part of the service or its dependencies.
+
+When [guard-gate](pkg/guard-gate) and a proper set of micro-rules are used, offenders will typically be required to build dedicated delivery mechanisms in order to explore options for detecting and exploiting service vulnerabilities. This lengthy process may need to be repeated for each service, as each service maintains a different set of micro-rules. As a result, an offender will not be able to use common statistical attack patterns.  
+
+Users of [guard-gate](pkg/guard-gate) gain Situational Awareness both thanks to alerts about out-of-pattern request/responses, and by the identification of security indicators that the service is misused. Such indicators include longer than usual  service times and the list of external IP addresses approached by the service.
+
+Beyond, Situational Awareness, [guard-gate](pkg/guard-gate) enables blocking out-of-pattern behaviors and the ability to react to potential attacks and/or to on-going attacks by introducing a fine-tune configurable security gate in front of the service.
+
+Overall, the solution offers both visibility into the security of the service, and the ability to monitor/block both known patterns and unknown patterns (using a zero-day exploits).
+
+<p align="center">
+    <img src="img/GuardInKnative.png" width="700"  />
+</p>
+
+## Solution Components
 
-## Solution components
 This project adds:
-1. [`guard-gate`](https:pkg/guard-gate) - the workload security gate
-1. A *"Guardian"*  - an object implemented as either a CRD or a Configmap maintaining a set of micro-rules per service 
-1. [`guard-learner`](https:cmd/guard-learner) - a learner service to auto learn the micro-rules stored in the *Guardian*  
-1. [`guard-ui`](https:cmd/guard-ui) - a user interface web app to simplify manual configuration of micro-rules 
+
+1. [guard-gate](pkg/guard-gate) - the workload security gate that is located in front of each service
+1. A **Guardian**  - an object implemented as either a CRD or a Configmap maintaining a set of micro-rules per service
```

Comment:
Any chance for a sequence diagram that shows how the vulnerability is trapped and handled. Essentially the components interactions...

Reply:
There is a diagram that shows the components and their interactions.